main/src/main/scala/sbt/Plugins.scala

```diff
@@ -345,4 +347,13 @@ ${listConflicts(conflicting)}""")
     }
     hasGetterOpt getOrElse false
   }
+
+  /** Debugging method to time how long it takes to run various compilation tasks. */
+  private[this] def timed[T](label: String, log: Logger)(t: => T): T = {
+    val start = System.nanoTime
+    val result = t
+    val elapsed = System.nanoTime - start
+    log.debug(label + " took " + (elapsed / 1e6) + " ms")
+    result
+  }
```

Comment:
We need to profile the difference, but the tricky part about `timed` is that it appears along the call graph at almost random places (megamorphic?) that JIT won't be able to inline them. It might actually be better if we turn it into a macro.